presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -356,16 +357,21 @@ public FunctionRegistry(TypeManager typeManager, BlockEncodingSerde blockEncodin
         // TODO the function map should be updated, so that this cast can be removed
         specializedScalarCache = CacheBuilder.newBuilder()
                 .maximumSize(1000)
+                // Old cached specialized functions might trigger some JVM bug, such as
+                // repeatedly compiling MethodHandle that causes full GC.
```

Comment:
```
We have observed repeated compilation of MethodHandle that lead to full GCs.
We notice that flushing the following caches mitigate the problem.
We suspect that it is a JVM bug that is related to stale/corrupted profiling data associated
with generated classes and/or dynamically-created MethodHandles.
This might also mitigate problems like deoptimization storm or unintended interpreted execution.